java/team25core/ObjectDetectionTask.java

```diff
@@ -0,0 +1,250 @@
+        EVERYTHING, //this may go away
+        OBJECT1_DETECTED,
+        OBJECT2_DETECTED,
+        LARGEST_SKY_STONE_DETECTED, //this may go away
+        UNKNOWN_DETECTED,
+    }
+     public enum ObjectKind {
+            OBJECT2_KIND,
+            OBJECT1_KIND,
+            UNKNOWN_KIND,
+     };
+
+
+    //for phone camera constructor
+    public ObjectDetectionTask(Robot robot)
+    {
+        super(robot);
+
+        rateLimitMs = 0;
+        detectionKind = DetectionKind.EVERYTHING;
+    }
+    //for webcamera construtor
+    public ObjectDetectionTask(Robot robot, String cameraName)
+    {
+        super(robot);
+        rateLimitMs = 0;
+        detectionKind = DetectionKind.EVERYTHING;
+        this.cameraName = cameraName;
+    }
+
+    private void initVuforia(HardwareMap hardwareMap) {
+        /*
+         * Configure Vuforia by creating a Parameter object, and passing it to the Vuforia engine.
+         */
+        //new=your own copy of vuforia parameters
+        VuforiaLocalizer.Parameters parameters = new VuforiaLocalizer.Parameters();
+        parameters.cameraDirection = VuforiaLocalizer.CameraDirection.BACK;
+        //if webcam
+        if (cameraName != null) {
+            parameters.vuforiaLicenseKey = VuforiaConstants.WEBCAM_VUFORIA_KEY;
+            parameters.cameraName = hardwareMap.get(WebcamName.class, cameraName);
+
+        } else {   // if phonecam
+            parameters.vuforiaLicenseKey = VuforiaConstants.VUFORIA_KEY;
+        }
+
+        //  Instantiate the Vuforia engine
+        vuforia = ClassFactory.getInstance().createVuforia(parameters);
+        Vuforia.setFrameFormat(PIXEL_FORMAT.RGB565,true);
+        vuforia.setFrameQueueCapacity(1);
+
+        // Loading trackables is not necessary for the Tensor Flow Object Detection engine.
+    }
+
+    private void initTfod(HardwareMap hardwareMap)
+    {
+        int tfodMonitorViewId = hardwareMap.appContext.getResources().getIdentifier(
+                "tfodMonitorViewId", "id", hardwareMap.appContext.getPackageName());
+        TFObjectDetector.Parameters tfodParameters = new TFObjectDetector.Parameters(tfodMonitorViewId);
+        tfodParameters.minResultConfidence = 0.8f; //the example in the Ultimate Goal Tensor flow example defaults to a MinimumConfidence of 0.8f
+        tfodParameters.isModelTensorFlow2 = true;
+        tfodParameters.inputSize = 320;
+
+        //concept tensor flow object detection had minimum confidence
+        tfod = ClassFactory.getInstance().createTFObjectDetector(tfodParameters, vuforia);
+        tfod.loadModelFromAsset(TFOD_MODEL_ASSET, LABEL_OBJECT1);
+    }
+
+    public void init(Telemetry telemetry, HardwareMap hardwareMap)
+    {
+        initVuforia(hardwareMap);
+
+        // if (ClassFactory.getInstance().canCreateTFObjectDetector()) {
+            initTfod(hardwareMap);
+//        } else {
+//            telemetry.addData("Sorry!", "This device is not compatible with TFOD");
+//        }
```

Comment:
Why is all of this commented out?  Does canCreateTFObjectDetector() return false for the Control Hub, which is where I'm presuming this is running?